Add kedro catalog resolve command (#2891)
kedro-org/kedro: docs/source/development/commands_reference.md

kedro catalog list --pipeline=ds,de
```

##### Resolve dataset factories in the catalog

```bash
kedro catalog resolve
```

> This command resolves dataset factories in the catalog file with any explicit entries in the pipeline. The output is the aggregation of all explicit dataset entries in the catalog and factory datasets in the pipeline.

Reviewer: I would say something like "This command shows a fully resolved catalog. It includes datasets explicitly mentioned in your catalog files and datasets mentioned in the project's pipelines resolved against corresponding dataset factory patterns". (Not exactly this but something to this effect)

> This command resolves dataset factories in the catalog file with any explicit entries in the pipeline. The output includes datasets explicitly mentioned in your catalog files and any datasets mentioned in the project's pipelines that resolve some dataset factory.

Reviewer 2: ```suggestion
This command resolves dataset factories in the catalog file with any explicit entries in the pipeline. The output includes datasets explicitly mentioned in your catalog files and any datasets mentioned in the project's pipelines that match a dataset factory.
```
